### PR TITLE
Refactor PowerShell install module

### PR DESCRIPTION
**Pull request by RassK**
## Why

Part of #1374

## What

Refactors some parts of the install module:

* Changed name: `install.psm1` -> `OpenTelemetry.Dotnet.Auto.psm1`
   * Adds proper module name according to PowerShell standards
   * Ensures there will be no conflicts with other modules named `install.psm1`
* Scoped all environment variables
* Added new global environment variable (`OTEL_DOTNET_AUTO_INSTALL_DIR`) for service location
* Added function to instrument current PowerShell session
* Fixes missing Temp directory (if it somehow got deleted)

## Checklist

- [x] Documentation is updated.

### CODE REVIEW

File: OpenTelemetry.Dotnet.Auto.psm1 [hunk 1]
```diff
@@ -287,9 +324,25 @@ function Unregister-OpenTelemetryForWindowsService() {
...
+        return $installDir
+    }
+
+    Write-Warning "OpenTelemetry .NET Automatic Instrumentation is not installed."
+}
+
 Export-ModuleMember -Function Install-OpenTelemetryCore
 Export-ModuleMember -Function Register-OpenTelemetryForIIS
 Export-ModuleMember -Function Register-OpenTelemetryForWindowsService
+Export-ModuleMember -Function Register-OpenTelemetryForCurrentSession
```
Comment (pellared) [inline OpenTelemetry.Dotnet.Auto.psm1:344]: Can you also add `Unregister-OpenTelemetryForCurrentSession`?

File: docs/README.md [hunk 2]
```diff
@@ -175,14 +175,20 @@ Register-OpenTelemetryForIIS
...
+
 # List all available commands
 Get-Command -Module install
 
 # Get command's usage information
 Get-Help Install-OpenTelemetryCore -Detailed
 ```
 
-⚠️ You have to reboot the machine
+⚠️ You have to restart the IIS / Windows service
```
Comment (pellared) [inline docs/README.md:191]: Maybe it could add the commands to the snippet as well?

```
Restart-Service -Name "MyServiceName"
& iisreset
```

Comment (RassK) [reply docs/README.md]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1473/commits/fe06b8f6dfbd72d3186e5bb138e7e7a8fbe44bce

File: OpenTelemetry.Dotnet.Auto.psm1 (hunk 1 above)
Comment (RassK) [reply OpenTelemetry.Dotnet.Auto.psm1:386]: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1473/commits/fe06b8f6dfbd72d3186e5bb138e7e7a8fbe44bce

File: docs/README.md (hunk 2 above)
Comment (RassK) [reply docs/README.md]: Warning also had to be changed >  https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1473/commits/9c78a6ff0fec6429f93b555d493771b5a3b42bec

File: docs/README.md [hunk 3]
```diff
@@ -175,15 +175,20 @@ Register-OpenTelemetryForIIS
...
 # List all available commands
 Get-Command -Module install
 
 # Get command's usage information
 Get-Help Install-OpenTelemetryCore -Detailed
 ```
 
-⚠️ You have to reboot the machine
-after setting up the instrumentation.
+⚠️ Registering for the IIS or a Windows service will perform a service restart!
```
Comment (pellared) [inline docs/README.md:191]:
```suggestion
⚠️ Registering for IIS and Windows Service will perform a service restart.
```

**Review by pellared — APPROVED** [review]

**Review by theletterf — APPROVED** [review]

**Pull request merged**